zboxcore/sdk/sdk.go

```diff
@@ -861,6 +861,19 @@ func CreateAllocationForOwner(name string, owner, ownerpublickey string,
 		return "", errors.New("", "invalid value for lock")
 	}
 
+	preferred, err := getPreferredBlobberIds(preferredBlobbers)
+	if err != nil {
+		return "", errors.New("failed_get_blobber_ids", "failed to get preferred blobber ids: "+err.Error())
+	}
+
+	allocationBlobbers, err := getAllocationBlobbers(owner, ownerpublickey, datashards,
+		parityshards, size, expiry, readPrice,
+		writePrice, mcct)
+	if err != nil {
+		return "", errors.New("failed_get_allocation_blobbers", "failed to get blobbers for allocation: "+err.Error())
+	}
+	ids := append(preferred, allocationBlobbers...)
```

Comment:
MY logic was that preferred blobbers are padded with allocation bobbers up to max_blobbers_per_allocation and then they are filtered in allocation creation, since only data_shards + parity_shards is needed. so I don't excessively validate preferred blobbers before two set merging. 
filtering duplicates though is a good idea, I'll implement it